browser/ui/webui/settings/brave_importer_observer_unittest.cc

```diff
@@ -0,0 +1,143 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/settings/brave_importer_observer.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/json/json_reader.h"
+#include "base/logging.h"
+#include "base/test/bind.h"
+#include "brave/browser/importer/brave_external_process_importer_host.h"
+#include "content/public/test/browser_task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+class BraveImporterObserverUnitTest : public testing::Test {
+ public:
+  BraveImporterObserverUnitTest() {}
+
+  void SetExpectedInfo(base::Value value) { expected_info_ = std::move(value); }
+  void SetExpectedCalls(int value) { expected_calls_ = value; }
+  int GetExpectedCalls() { return expected_calls_; }
+  void NotifyImportProgress(const importer::SourceProfile& source_profile,
+                            const base::Value& info) {
+    EXPECT_EQ(expected_info_, info);
+    expected_calls_++;
+  }
+
+ private:
+  content::BrowserTaskEnvironment task_environment_;
+  base::Value expected_info_;
+  int expected_calls_ = 0;
+  raw_ptr<BraveExternalProcessImporterHost> _ = nullptr;
+};
+
+TEST_F(BraveImporterObserverUnitTest, ImportEvents) {
+  auto* importer_host = new BraveExternalProcessImporterHost();
+  importer::SourceProfile source_profile;
+  source_profile.importer_name = u"importer_name";
+  source_profile.importer_type = importer::TYPE_CHROME;
+  source_profile.source_path = base::FilePath(FILE_PATH_LITERAL("test"));
+  auto imported_items = importer::AUTOFILL_FORM_DATA | importer::PASSWORDS;
+  std::unique_ptr<BraveImporterObserver> observer =
+      std::make_unique<BraveImporterObserver>(
+          importer_host, source_profile, imported_items,
+          base::BindRepeating(
+              &BraveImporterObserverUnitTest::NotifyImportProgress,
+              base::Unretained(this)));
+  importer_host->set_observer(observer.get());
```

Comment:
nit: this can be deleted as it's done by observer.

Reply:
done